pkg/worker/k8srunner.go

```diff
@@ -252,17 +252,27 @@ func (r k8sStepRunner) waitForPodModifiedFunc(ctx context.Context, podMeta metav
 	}
 	defer w.Stop()
 	for ev := range w.ResultChan() {
-		pod := ev.Object.(*v1.Pod)
-		switch ev.Type {
-		case watch.Modified:
-			ok, err := f(pod)
-			if err != nil {
-				return err
-			} else if ok {
-				return nil
+		switch ev.Object.(type) {
+		case *v1.Pod:
+			pod := ev.Object.(*v1.Pod)
```

Comment:
nit: You can assign a variable directly in a type-switch statement. Suggest doing that, as it reduces the type-cast duplication

Like so:

```go
switch obj := ev.Object.(type) {
case *v1.Pod:
	// obj is of type *v1.Pod
case *metav1.Status:
	// obj is of type *metav1.Status
}
```